packages/mui-material/src/styles/props.d.ts

```diff
@@ -179,7 +179,7 @@ export interface ComponentsPropsList {
   MuiInputBase: InputBaseProps;
   MuiInputLabel: InputLabelProps;
   MuiLinearProgress: LinearProgressProps;
-  MuiLink: LinkProps;
+  MuiLink: LinkProps<React.ElementType>;
```

Comment:
Is this required?

Reply:
It was added to remove hacks like i did here https://github.com/mui/material-ui/pull/35924#discussion_r1164500104, michal was ok to convert it to generic but needed 2nd opinion. related comment: https://github.com/mui/material-ui/pull/35924#discussion_r1192419467